README.md

```diff
@@ -1,5 +1,175 @@
 <a href="https://www.npmjs.com/package/@paladin/stake"><img src="https://img.shields.io/npm/v/%40paladin%2Fstake?logo=npm&color=377CC0" /></a>
 <a href="https://crates.io/crates/paladin-stake"><img src="https://img.shields.io/crates/v/paladin-stake?logo=rust" /></a>
+
+The Paladin Stake program manages the delegation of tokens to a particular validator of the system. This allows stakers to earn additional shares of rewards proportional to their share of staked tokens and to participate in governance.
+
+## Overview
+
+A `Config` account essentailly represents the staking system and it is the first account that needs to be created. Once a `Config` account is created, `ValidatorStake` accounts can be added to the system. A `ValidatorStake` represents a validator of the network (`VoteState` account). Anybody can create the stake account for a validator. For new accounts, the authority is initialized to the validator vote account's withdraw authority.
+
+After `ValidatorStake` accounts are added to the system, `SolStakerStake` accounts can be created. These accounts represent individual SOL stakers (`StakeState` accounts). Similarly to `ValidatorStake`, anybody can create the stake account for a SOL staker. For new accounts, the authority is initialized to the stake state account's withdrawer. When stake is added to a `SolStakerStake`, it will update the total amount of staked tokens both on the correspodent `ValidatorStake` and `Config`. Therefore each `ValidatorStake` tracks the amount of staked tokens that its individual stakers are currently staking, while the `Config` tracks the total of staked tokens on the system. The system also keeps track of the SOL amount staked on the network to determine the share of the rewards and enforce that the proportion of tokens and SOL staked are within the expected limits.
```

Comment:
nit: "stake state" isn't very clear
```suggestion
After `ValidatorStake` accounts are added to the system, `SolStakerStake` accounts can be created. These accounts represent individual SOL stakers (coupled to `StakeState` accounts in the native Solana stake program). Similarly to `ValidatorStake`, anybody can create the stake account for a SOL staker. For new accounts, the authority is initialized to the SOL stake account's withdrawer. When stake is added to a `SolStakerStake`, it will update the total amount of staked tokens both on the corresponding `ValidatorStake` and `Config`. Therefore each `ValidatorStake` tracks the amount of staked tokens that its individual stakers are currently staking, while the `Config` tracks the total of staked tokens on the system. The system also keeps track of the SOL amount staked on the network to determine the share of the rewards and enforce that the proportion of tokens and SOL staked are within the expected limits.
```